src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,121 @@
+                        // data in textfile contains : after at, to replace it with whitespace
+                        String modEventTime = eventTime.replaceFirst(": ", " ");
+                        Event newTask = new Event(description, modEventTime);
+                        newTask.markAsDone();
+                        tasklist.add(newTask);
+                    }
+                } else if (modifiedLine.contains("[D]")) {
+                    if (modifiedLine.contains("[ ]")) {
+                        // replace the formats with empty string
+                        String deadlineDsc = modifiedLine.replace("[D][ ] ", "");
+                        // get 2 substring for description and deadlineTime
+                        String description = deadlineDsc.substring(0, deadlineDsc.indexOf("(") - 1);
+                        String dlTime = deadlineDsc.substring(deadlineDsc.indexOf("(")+1,deadlineDsc.indexOf(")"));
+                        String modDeadlineTime = dlTime.replaceFirst("by: ", "");
+                        // to convert string to date
+                        DateTimeFormatter convertFormatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+                        LocalDate deadlineDate = LocalDate.parse(modDeadlineTime, convertFormatter);
+                        // to put in the same format as Deadline class parameter
+                        DateTimeFormatter stringFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                        String modifiedString = deadlineDate.format(stringFormatter);
+                        Deadline newTask = new Deadline(description, modifiedString);
+                        tasklist.add(newTask);
+                    } else {
+                        // replace the formats with empty string
+                        String deadlineDsc = modifiedLine.replace("[D][X] ", "");
+                        // get 2 substring for description and deadlineTime
+                        String description = deadlineDsc.substring(0, deadlineDsc.indexOf("(") - 1);
+                        String dlTime = deadlineDsc.substring(deadlineDsc.indexOf("(")+1,deadlineDsc.indexOf(")"));
+                        String modDeadlineTime = dlTime.replaceFirst("by: ", "");
+                        // to convert string to date
+                        DateTimeFormatter convertFormatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+                        LocalDate deadlineDate = LocalDate.parse(modDeadlineTime, convertFormatter);
+                        // to put in the same format as Deadline class parameter
+                        DateTimeFormatter stringFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                        String modifiedString = deadlineDate.format(stringFormatter);
+                        Deadline newTask = new Deadline(description, modifiedString);
+                        newTask.markAsDone();
+                        tasklist.add(newTask);
```

Comment:
I like the use of comments to illustrate your code which improves readability. 👍